internal/cmd/beta/network/create/create.go

```diff
@@ -0,0 +1,178 @@
+			),
+		),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			ctx := context.Background()
+			model, err := parseInput(p, cmd)
+			if err != nil {
+				return err
+			}
+
+			// Configure API client
+			apiClient, err := client.ConfigureClient(p)
+			if err != nil {
+				return err
+			}
+
+			projectLabel, err := projectname.GetProjectName(ctx, p, cmd)
+			if err != nil {
+				p.Debug(print.ErrorLevel, "get project name: %v", err)
+				projectLabel = model.ProjectId
+			}
+
+			if !model.AssumeYes {
+				prompt := fmt.Sprintf("Are you sure you want to create a network for project %q?", projectLabel)
+				err = p.PromptForConfirmation(prompt)
+				if err != nil {
+					return err
+				}
+			}
+
+			// Call API
+			req := buildRequest(ctx, model, apiClient)
+			resp, err := req.Execute()
+			if err != nil {
+				return fmt.Errorf("create network : %w", err)
+			}
+			networkId := *resp.NetworkId
+
+			// Wait for async operation, if async mode not enabled
+			if !model.Async {
+				s := spinner.New(p)
+				s.Start("Creating network")
+				_, err = wait.CreateNetworkWaitHandler(ctx, apiClient, model.ProjectId, networkId).WaitWithContext(ctx)
+				if err != nil {
+					return fmt.Errorf("wait for network creation: %w", err)
+				}
+				s.Stop()
+			}
+
+			return outputResult(p, model, projectLabel, resp)
+		},
+	}
+	configureFlags(cmd)
+	return cmd
+}
+
+func configureFlags(cmd *cobra.Command) {
+	cmd.Flags().StringP(nameFlag, "n", "", "Network name")
+	cmd.Flags().StringSlice(dnsServersFlag, []string{}, "List of DNS servers/nameservers IPs")
+	cmd.Flags().Int64(prefixLengthFlag, 0, "The default prefix length for networks")
+
+	err := flags.MarkFlagsRequired(cmd, nameFlag)
+	cobra.CheckErr(err)
+}
+
+func parseInput(p *print.Printer, cmd *cobra.Command) (*inputModel, error) {
+	globalFlags := globalflags.Parse(p, cmd)
+	if globalFlags.ProjectId == "" {
+		return nil, &cliErr.ProjectIdError{}
+	}
+
+	model := inputModel{
+		GlobalFlagModel: globalFlags,
+		Name:            flags.FlagToStringPointer(p, cmd, nameFlag),
+		DnsServers:      flags.FlagToStringSlicePointer(p, cmd, dnsServersFlag),
+		PrefixLength:    flags.FlagToInt64Pointer(p, cmd, prefixLengthFlag),
+	}
+
+	if p.IsVerbosityDebug() {
+		modelStr, err := print.BuildDebugStrFromInputModel(model)
+		if err != nil {
+			p.Debug(print.ErrorLevel, "convert model to string for debugging: %v", err)
+		} else {
+			p.Debug(print.DebugLevel, "parsed input values: %s", modelStr)
+		}
+	}
+
+	return &model, nil
+}
+
+func buildRequest(ctx context.Context, model *inputModel, apiClient *iaas.APIClient) iaas.ApiCreateNetworkRequest {
+	req := apiClient.CreateNetwork(ctx, model.ProjectId)
+
+	payload := iaas.CreateNetworkPayload{
+		Name: model.Name,
+		AddressFamily: &iaas.CreateNetworkAddressFamily{
+			Ipv4: &iaas.CreateNetworkIPv4{
+				Nameservers:  model.DnsServers,
+				PrefixLength: model.PrefixLength,
+			},
+		},
+	}
```

Comment:
The API seems to allow some more fields, e.g. `Ipv6` in the `AddressFamily`, `routed` at the root level, `gateway` inside `Ipv4` or `Ipv6`. Shouldn't we also have those in the CLI?

Reply:
I think we should add those new fields. They were not in refinement, so I did not add but refinement might be outdated after api changes.